pkg/remotewrite/config.go

```diff
@@ -1,108 +1,85 @@
 package remotewrite
 
 import (
+	"crypto/tls"
 	"encoding/json"
-	"net/url"
+	"net/http"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/grafana/xk6-output-prometheus-remote/pkg/remote"
 	"github.com/kubernetes/helm/pkg/strvals"
-	promConfig "github.com/prometheus/common/config"
-	"github.com/prometheus/common/model"
-	"github.com/prometheus/prometheus/storage/remote"
+
 	"go.k6.io/k6/lib/types"
 	"gopkg.in/guregu/null.v3"
 )
 
 const (
-	defaultPrometheusTimeout = time.Minute
-	defaultFlushPeriod       = 5 * time.Second
-	defaultMetricPrefix      = "k6_"
+	defaultURL          = "http://localhost:9090/api/v1/write"
+	defaultTimeout      = 5 * time.Second
+	defaultPushInterval = 5 * time.Second
+	defaultMetricPrefix = "k6_"
 )
 
 type Config struct {
-	Mapping null.String `json:"mapping" envconfig:"K6_PROMETHEUS_MAPPING"`
-
-	Url null.String `json:"url" envconfig:"K6_PROMETHEUS_REMOTE_URL"` // here, in the name of env variable, we assume that we won't need to distinguish between remote write URL vs remote read URL
-
+	// here, in the name of env variable, we assume that we won't need to distinguish between remote write URL vs remote read URL
+	URL     null.String       `json:"url" envconfig:"K6_PROMETHEUS_REMOTE_URL"`
```

Comment:
This comment makes me think there is some read functionality?

Reply:
It is, the Remote write protocol defines the endpoints for reading and for writing. Now, you're asking I'm thinking if we should break change and just have the path added as a constant from the `WriteClient`.

So the user instead passes:
```
K6_PROMETHEUS_REMOTE_URL=http://localhost:9090/api/v1/write
```
will pass:
```
K6_PROMETHEUS_REMOTE_URL=http://localhost:9090/api/v1/write
```

Potentially, we could also do both so use the explicit path if set otherwise add it but I don't know if we should support this complexity without an explicit feature request from users.

@grafana/k6-devs WDYT?

Reply:
@codebien, But how likely is it possible that in this output extension, we will ever use the read endpoint? :thinking: I agree with this, and it seems that this comment can probably lead to some confusion, and its value is questionable.

>WDYT?

IMO, the first option (where we define the endpoint `http://localhost:9090/api/v1/write`) still looks a bit more flexible and preferable as long as we not planning to use anything apart from that endpoint.

Reply:
I rechecked the specs and the path is not fixed, I was convinced the protocol defined the path but this is obviously not true. The path depends on the implementation and it is configurable so we have to get it explicitly.

Reply:
I added documentation for these fields so we can now remove the original comment.